pkg/asset/agent/image/oc.go

```diff
@@ -88,19 +89,29 @@ func (r *release) GetBaseIso(log logrus.FieldLogger, releaseImage string, pullSe
 	return path, err
 }
 
-func (r *release) getImageFromRelease(log logrus.FieldLogger, imageName, releaseImage, pullSecret string, haveMirror bool) (string, error) {
+func (r *release) getImageFromRelease(log logrus.FieldLogger, imageName, releaseImage, pullSecret string, mirrorConfig []mirror.RegistriesConfig) (string, error) {
 	// This requires the 'oc' command so make sure its available
 	_, err := exec.LookPath("oc")
 	if err != nil {
-		if haveMirror {
+		if len(mirrorConfig) > 0 {
 			log.Warning("Unable to validate mirror config because \"oc\" command is not available")
 		} else {
 			log.Debug("Skipping ISO extraction; \"oc\" command is not available")
 		}
 		return "", err
 	}
 
-	cmd := fmt.Sprintf(templateGetImage, imageName, true, releaseImage)
+	if len(mirroConfig) > 0 {
```

Comment:
Any specific reason to create/delete twice the icsp file? It could be generated just once at the beginning of `GetBaseIso` IINW

Reply:
and add it to the signature of the other two functions? It can be done.